kubernetes/linux/main.sh

```diff
@@ -334,32 +352,44 @@ if [ -e "/etc/ama-logs-secret/WSID" ]; then
       fi
 
       if [ $? -ne 0 ]; then
-            if [ ! -z "$PROXY_ENDPOINT" ]; then
-               if [ -e "/etc/ama-logs-secret/PROXYCERT.crt" ]; then
-                  echo "Making curl request to ifconfig.co with proxy and proxy CA cert"
-                  RET=`curl --max-time 10 -s -o /dev/null -w "%{http_code}" ifconfig.co --proxy $PROXY_ENDPOINT --proxy-cacert /etc/ama-logs-secret/PROXYCERT.crt`
-               else
-                  echo "Making curl request to ifconfig.co with proxy"
-                  RET=`curl --max-time 10 -s -o /dev/null -w "%{http_code}" ifconfig.co --proxy $PROXY_ENDPOINT`
-               fi
+            registry="https://mcr.microsoft.com/v2/"
+            if [ $CLOUD_ENVIRONMENT == "azurechinacloud" ]; then
+                  registry="https://mcr.azk8s.cn/v2/"
+            fi
+            if [ $CLOUD_ENVIRONMENT == "usnat" ] || [ $CLOUD_ENVIRONMENT == "ussec" ]; then
+                  registry=$MCR_URL
+            fi
+            if [ -z $registry ]; then
+                  echo "The environment variable MCR_URL is not set for CLOUD_ENVIRONMENT: $CLOUD_ENVIRONMENT"
+                  RET=000
             else
-                  echo "Making curl request to ifconfig.co"
-                  RET=$(curl --max-time 10 -s -o /dev/null -w "%{http_code}" ifconfig.co)
+                  if [ ! -z "$PROXY_ENDPOINT" ]; then
```

Comment:
There is indention issue. Block under if [ ! -z "$PROXY_ENDPOINT" ]; then needs to be intended correctly.